Wilab-io/sample-apps
# [Image search] Develop image search, results list and refactor the settings screen

# Description

I developed the image search feature, the results list screen and refactored the settings to allow users to navigate through tabs freely.

# How to test

1. Go to your application's console: https://console.vespa-cloud.com/tenant/{tenant_name}/application/{app_name}/prod/deploy
and download the app key certificate into this app's `application` directory.

<img width="745" alt="image" src="https://github.com/user-attachments/assets/58dadf8c-4a95-4ef5-a792-256820bf2b5b">
<img width="346" alt="image" src="https://github.com/user-attachments/assets/206be000-5cf9-492e-aea9-8dcecafec61e">

2. Set up a Postgres database with this configuration: `postgres://postgres:postgres@127.0.0.1:5432/postgres`. [Example here](https://medium.com/@vishal.sharma./run-postgresql-and-pgadmin-using-docker-compose-34120618bcf9). [More info here](https://stackoverflow.com/questions/3582552/what-is-the-format-for-the-postgresql-connection-string-url).
3. Then go to the `src` directory and run `python main.py`.
4. When the app is up, you can log in with admin:1 or demo:1 users.
5. Upload at least one document.
6. Go through the configuration pages and fill in every required field.
7. The deploy button should now be activated, click it and follow the extra login step when prompted before hitting the "Continue" button.
8. Once you see the "Deployment successful" modal you can start searching your documents in the home page.


Closes #40
Closes #41 
Closes #42
